datacommons_client/endpoints/node.py

```diff
@@ -252,99 +253,233 @@ def fetch_entity_names(
 
     return names
 
+  def _fetch_contained_in_place(
+      self,
+      node_dcids: str | list[str],
+      out: bool = True,
+      contained_type: Optional[str] = None,
+      as_dict: bool = False,
+  ) -> dict[str, list[Node | dict]]:
+    """Fetches places that contain or are contained in the given nodes. Uses the
+          `containedInPlace` property to fetch parent or child place relationships.
+
+        Args:
+            node_dcids (str | list[str]): One or more DCIDs representing geographic places.
+            out (bool, optional): If True, fetch places contained in the given node(s).
+                If False, fetch places that contain the given node(s). Defaults to True.
+            contained_type (str, optional): Optional type constraint (e.g., 'Country',
```

Comment:
I believe `contained_type` may only be specified when out=False, otherwise the REST api gives an error: https://api.datacommons.org/v2/node?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI&nodes=geoId%2F06&property=-%3EcontainedInPlace%2B%7BtypeOf%3ACountry%7D

even though it's just a helper method, it may be worth adding a param validation check in this just to make it clear in case someone else builds on it.